cg/models/downsample/downsample_data.py

```
@@ -0,0 +1,161 @@
+        self,
+    ) -> Sample:
+        """
+        Generate a downsampled sample record for StatusDB.
+        The new sample contains the original sample internal id and meta data.
+        Raises:
+            ValueError if the downsampled sample already exists in statusDB
+        """
+        application_version: ApplicationVersion = self.get_application_version(self.original_sample)
+        downsampled_sample: Sample = self.status_db.add_sample(
+            name=self.downsampled_sample_name,
+            internal_id=self.downsampled_sample_name,
+            sex=self.original_sample.sex,
+            order=self.original_sample.order,
+            reads=multiply_by_million(self.number_of_reads),
+            downsampled_to=multiply_by_million(self.number_of_reads),
+            from_sample=self.original_sample.internal_id,
+            tumour=self.original_sample.is_tumour,
+            priority=Priority.standard,
+            customer=self.original_sample.customer,
+            application_version=application_version,
+        )
+        return downsampled_sample
+
+    def _generate_statusdb_downsampled_case(
+        self,
+    ) -> Family:
+        """Generate a case for the downsampled samples. The new case uses existing case data."""
+        if case_exists_in_statusdb(status_db=self.status_db, case_name=self.downsampled_case_name):
+            return self.status_db.get_case_by_name(self.downsampled_case_name)
+        downsampled_case: Family = self.status_db.add_case(
+            data_analysis=self.original_case.data_analysis,
+            data_delivery=self.original_case.data_delivery,
+            name=self.downsampled_case_name,
+            panels=self.original_case.panels,
+            priority=self.original_case.priority,
+            ticket=self.original_case.latest_ticket,
+        )
+        downsampled_case.customer = self.original_case.customer
+        return downsampled_case
+
+    def has_enough_reads_to_downsample(self) -> bool:
```

If it raises an error instead of returning false when it does not have enough reads, I suggest renaming this function to validate_has_enough_reads, and making it return None